src/ngraph_builder.cc

```diff
@@ -41,6 +41,23 @@ const static std::map<tf::DataType, ngraph::element::Type> TF_NGRAPH_TYPE_MAP =
      {tf::DataType::DT_UINT16, ng::element::u16},
      {tf::DataType::DT_BOOL, ng::element::boolean}};
 
+// Helper for storing ops in ng_op_map
+static void SaveNgOp(Builder::OpMap& map, const std::string key,
+                     const shared_ptr<ng::Node> value) {
+  map[key].push_back(value);
+}
+
+// Helper for lookup in ng_op_map
+static tf::Status GetNgOp(const Builder::OpMap& map, std::string key,
```

Comment:
The defaulting behavior here won't work. We never know without checking the edge that an incoming input came from the source node's 0th output. Instead I think we will have to switch from using `Node::input_node` to using `Node::input_edge` (or perhaps `Node::input_edges` for some ops) so that we know which output index to query.

Reply:
I do not fully understand the problem - the default behavior reflects previous(master) code version - where `ng_op_map` maps s`tring->ng::Node`. Now we have `string->vector<ng::Node>`. If we use default (index=0) version - it behaves like the map `string->ng::Node` - fetching first element from vector. 

For each input, there is a different key in the map, 
for example some Pack op has three inputs with names:
`ngraph_input_0, ngraph_input_1, ngraph_input_2`
being fetched from `ng_op_map` separately one by one (all at index 0) 
`GetNgOp(ng_op_map, ngraph_input_X, result (index =0))`

One `output_node` is created and saved
`SaveNgOp(ng_op_map, ‘pack’, output_node)`

Now, `vector` was added in order to save multiple output nodes in the map.

So, for example, Split node has three outputs, calling 
`SaveNgOp(ng_op_map, ‘split’, output_node_0)`
`SaveNgOp(ng_op_map, ‘split’, output_node_1)`
`SaveNgOp(ng_op_map, ‘split’, output_node_2)`

So at key ‘split’ we now have vector of 3.

At the end of `TranslateGraph` function, there is:

` for (auto n : tf_ret_vals) {
   ….

    int index;
    if (tf::GetNodeAttr(n->attrs(), "index", &index) != tf::Status::OK()) {
      return tf::errors::InvalidArgument("No index defined for _Retval");
    }

    shared_ptr<ng::Node> result;

    tf::Status status =
          GetNgOp(ng_op_map, tf_input_node->name(), result, index);
    ng_result_list[index] = result; 
`

So we are saving in the same order as it was written to ng_op_map - is this a problem here ?

Reply:
Got it. Also, I've checked that tf output order depends on the order the tensors are used in next ops, so we have to address two things:
- when saving op results, write it in correct order
- when fetching input_node, get the particular output(s) in correct order
Code
```    (t0,t1) = tf.split(a, [2,3], 0)
    t1plus = t1 + [0,0,0]
    t0plus = t0 + [9,8]
    (tx0,tx1) = tf.split(a, [2,3], 0)
    tx0plus = tx0 + [19,18]
    tx1plus = tx1 + [10,10,10]
```
Generated following graph - see the output order for split
![screen shot 2018-07-24 at 5 29 53 pm](https://user-images.githubusercontent.com/11440049/43149236-45154de0-8f67-11e8-8105-cccb23896d94.png)

Reply:
I think the easiest way to resolve this would be to use op->in_edges() and op->out_edges() as the information which inputs to fetch/save.